### PR TITLE
Custom python script to insert user-supplied root CAs and mount points

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## What does this PR do? 

This is probably one of the more...creative things I've done. This is my second attempt at creating some kind of script that would allow the users to point the plugin at a custom root cert. From what I can tell, no other plugins really attempt to solve this problem. This particular solution meets a few requirements I had in mind:

- Relatively small build-time requirements for the user
- Doesn't require us to host any additional build artifacts

This is a (relatively) small python script that uses the Docker Registry API to find the plugin, manually download the raw plugin artifacts, and build the plugin. This is actually rather similar to what happens under the hood when you run `docker plugin install`, we just do it ourselves so we can insert user config. Despite the fact that docker plugins and "normal" containers are stored in the Registry in a nearly identical way, we can't run `docker pull` on a plugin.

From the docs:

You can also specify a custom name and version tag for the rebuild plugin via `--tag` and `--name`. By default, the rebuilt plugin will iterate the patch release of the latest plugin.
By defualt, this script will build a new plugin based on the latest release available from the docker store. To specify a different version, use the `--from_release` argument.

This script rebuilds a plugin and adds a custom bindmount path. This can be used to add user-supplied certificates to a Plugin via a mounted directory. 
The host directory will be mounted as `/usr/local/share/ca-certificates` inside the plugin.

After the bindmount has been created, you can adjust the bindmount source and destination via `docker plugin set`:

`$ docker plugin set elastic/elastic-logging-plugin:8.0.0 CUSTOM_DIR.source="/new/source/bindmount"`

In cases where you're building the plugin on a system that doesn't have access to the docker daemon where you want to run the plugin, 
you can either use `docker plugin push` to push to your own docker repo (note that you must change the name with `--name`), or use the `--to_tar` flag with a target directory to generate a tarball.
For example: `./setmount.py --to_tar /path/to/write/tar/new_plugin.tar.gz /path/to/certs/on/docker/host`
To generate a docker plugin from the tarball, unpack the `.tar.gz` archive, and then run `docker plugin create NEW_PLUGIN_NAME PATH_TO_UNPACKED_DIRECTORY`.

My python is a tad rusty, so forgive me if there's a few stylistic errors here.

## Why is it important?

Many users will have Elastic Stack setups that require a custom cert.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ] Add user-facing docs on usage

## How to test this PR locally

- pull down PR, `cd` to `tools` directory.
- Install the `requests` library if you don't have it already.
- run `./setmount.py /path/to/certs/on/docker/host`
- run `./setmount.py --to_tar /path/to/certs/on/docker/host` to test tar writing
- run `docker plugin inspect ...` to check for the presence of the mount specified above
- enable or install the plugin, run a container with the plugin.

### CODE REVIEW

**Comment by kvch** [comment]
May I ask why you opted for writing the script in Python? AFAIK we are trying to get rid of Python scripts where possible.

**Comment by fearful-symmetry** [comment]
@kvch Two things

- Doesn't require us to host any binary artifacts 
- Probably more likely to be installed then `mage`

**Comment by kvch** [comment]
How would you distribute it? Do we expect users to fetch the script from this repo? Is the problem you are solving a common user issue? Is it for development? I am sorry for the many questions. I am trying to understand this PR and your decisions. :)

In general, I would rather host a binary which can be run in any environment. My problem with Python scripts that users have to install its dependencies and need the appropriate version of the interpreter. More things can go wrong with such scripts than with binaries. Storage/hosting binaries cost less than engineering hours trying to track down issues. :D

**Comment by fearful-symmetry** [comment]
@kvch A small number of users are probably going to use this, and my thinking was to support a relatively uncommon use case without creating another set of build artifacts at risk of neglect.

File: x-pack/dockerlogbeat/tools/setmount.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    cleanup(tmpfs)
+
+
+def rebuild_plugin(tmpfs: str, name: str, tag: str):
+    """Rebuild the plugin via docker"""
+
+    # so we don't collide we anything else, iterate the semver string
+    reg = re.compile(r'''(\d+)\.(\d+)\.(\d+)''')
+    semver = list(reg.findall(tag))[0]
+    if len(semver) is not 3:
```
Comment (masci) [inline x-pack/dockerlogbeat/tools/setmount.py:71]: `is` compares objects, better to use `!=` here (Python 3.8+ will output a warning)

File: x-pack/dockerlogbeat/tools/setmount.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+def get_tag(session: requests.Session, from_release: str):
+    """Get the correct tag. 'latest' doesn't seem to play nice with plugins."""
+
+    tags_url = API_URL + "/" + PLUGIN_URI + "/tags/list"
+    tags_resp = session.get(tags_url)
+    tags_resp.raise_for_status()
+    tags = tags_resp.json()["tags"]
+    if from_release in tags:
+        return from_release
+    if from_release is not "" and from_release not in tags:
```
Comment (masci) [inline x-pack/dockerlogbeat/tools/setmount.py:179]: Same here, let's do `if from_release != ""`

**Review by masci — COMMENTED** [review]
Did you consider adding a README to the `tools` folder with some content from this PR description, explaining what problem the script solves and how to run it (install dependencies and other system requirements)? That might help users discover the tool as I assume there will be no further documentation for this.

**Comment by fearful-symmetry** [comment]
@masci Yep! I haven's started on the documentation yet, I wanted to run the code by people and then work on the asciidocs.

**Comment by jlind23** [comment]
@fearful-symmetry - Closing this one as there were no activity for a while

**Pull request closed without merging**